build-tools/packages/build-cli/src/commands/generate/sourceEntrypoints.ts

```diff
@@ -0,0 +1,410 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import fs from "node:fs/promises";
+import { Flags } from "@oclif/core";
+import JSON5 from "json5";
+import type { ExportSpecifierStructure, Node } from "ts-morph";
+import { ModuleKind, Project, ScriptKind } from "ts-morph";
+
+import type { PackageJson } from "@fluidframework/build-tools";
+
+import { ApiLevel, ApiTag, BaseCommand, isKnownApiLevel } from "../../library/index.js";
+import {
+	readPackageJson,
+	// AB#8118 tracks removing the barrel files and importing directly from the submodules, including disabling this rule.
+	// eslint-disable-next-line import/no-internal-modules
+} from "../../library/package.js";
+// AB#8118 tracks removing the barrel files and importing directly from the submodules, including disabling this rule.
+// eslint-disable-next-line import/no-internal-modules
+import { type ExportData, getExportPathFromPackage } from "../../library/packageExports.js";
+// AB#8118 tracks removing the barrel files and importing directly from the submodules, including disabling this rule.
+// eslint-disable-next-line import/no-internal-modules
+import { getApiExports, getPackageDocumentationText } from "../../library/typescriptApi.js";
+import type { CommandLogger } from "../../logging.js";
+
+/**
+ * Generates source entrypoints for Fluid Framework APIs to support API levels (/alpha, /beta. etc.).
+ */
+export default class GenerateSourceEntrypointsCommand extends BaseCommand<
+	typeof GenerateSourceEntrypointsCommand
+> {
+	static readonly description =
+		`Generates TypeScript source files that roll up APIs into different entrypoint files, defined by the "exports" field in package.json and organized by API tags`;
+
+	static readonly flags = {
+		mainEntrypoint: Flags.file({
+			description: "Main entrypoint file containing all untrimmed exports.",
+			default: "./src/index.ts",
+			exists: true,
+		}),
+		outDir: Flags.directory({
+			description: "Directory to emit entrypoint files.",
+			default: "./src/entrypoints/",
+			exists: true,
+		}),
```

Comment:
While this description might be true the only use of this now is to find tsconfig file. This really is the path to directory that contains a file named "tsconfig.json". (I recommend changing it to be explicitly full tsconfig path. Much more flexible and makes the requirement clear.)